src/capabilities.rs

```diff
@@ -0,0 +1,189 @@
+
+use crate::{
+    capi,
+    storage::StorageCapabilities,
+    camera::CameraCapabilities,
+    components::{macros::impl_plain_old_dict, Property},
+};
+
+#[pyclass]
+#[derive(Clone, Serialize, Deserialize)]
+pub struct VideoStreamCapabilities {
+    #[pyo3(get, set)]
+    camera: Py<CameraCapabilities>,
+
+    #[pyo3(get, set)]
+    storage: Py<StorageCapabilities>,
+
+    #[pyo3(get, set)]
+    max_frame_count: Py<Property>,
+
+    #[pyo3(get, set)]
+    frame_average_count: Py<Property>,
+}
+
+impl_plain_old_dict!(VideoStreamCapabilities);
+
+impl Default for VideoStreamCapabilities {
+    fn default() -> Self {
+        Python::with_gil(|py| Self {
+            camera: Py::new(py, CameraCapabilities::default()).unwrap(),
+            storage: Py::new(py, StorageCapabilities::default()).unwrap(),
+            max_frame_count: Py::new(py, Property::default()).unwrap(),
+            frame_average_count: Py::new(py, Property::default()).unwrap(),
+        })
+    }
+}
+
+impl TryFrom<capi::AcquirePropertyMetadata_aq_metadata_video_s> for VideoStreamCapabilities {
+    type Error = anyhow::Error;
+
+    fn try_from(
+        value: capi::AcquirePropertyMetadata_aq_metadata_video_s,
+    ) -> Result<Self, Self::Error> {
+        Ok(Python::with_gil(|py| -> PyResult<_> {
+            let camera: CameraCapabilities = value.camera.try_into()?;
+            let storage: StorageCapabilities = value.storage.try_into()?;
+            let max_frame_count: Property = value.max_frame_count.try_into()?;
+            let frame_average_count: Property = value.frame_average_count.try_into()?;
+
+            Ok(Self {
+                camera: Py::new(py, camera)?,
+                storage: Py::new(py, storage)?,
+                max_frame_count: Py::new(py, max_frame_count)?,
+                frame_average_count: Py::new(py, frame_average_count)?,
+            })
+        })?)
+    }
+}
+
+#[pyclass]
+#[derive(Clone, Serialize, Deserialize)]
+pub struct Capabilities {
+    #[pyo3(get, set)]
+    video: (Py<VideoStreamCapabilities>, Py<VideoStreamCapabilities>),
+}
+
+#[pymethods]
+impl Capabilities {
+    #[new]
+    #[pyo3(signature = (**kwargs))]
+    fn __new__(kwargs: Option<&pyo3::types::PyDict>) -> anyhow::Result<Self> {
+        if let Some(kwargs) = kwargs {
+            Ok(pythonize::depythonize(kwargs)?)
+        } else {
+            Ok(Default::default())
+        }
+    }
+
+    fn dict(&self, py: Python<'_>) -> PyResult<Py<PyAny>> {
+        Ok(pythonize::pythonize(py, self)?)
+    }
+
+    fn __repr__(&self, py: Python<'_>) -> PyResult<String> {
+        let obj = pythonize::pythonize(py, self)?;
+        let obj = obj.as_ref(py).downcast::<pyo3::types::PyDict>()?;
+        let args: String = obj
+            .iter()
+            .map(|(k, v)| format!("{}='{}'", k, v))
+            .reduce(|acc, e| format!("{},{}", acc, e))
+            .unwrap_or(String::new());
+
+        Ok(format!("Capabilities({})", args))
+    }
```

Comment:
Why do we need these explicit implementations for `Capabilities`, but not for other structs?

Reply:
I just copied this over from `Properties`. The `impl_plain_old_dict` macro does the same thing. It's not a big deal to change.